src/crosschain/zappingOoki.ts

```diff
@@ -0,0 +1,119 @@
+import { MulticallRouter, Ooki } from './contracts'
+
+export class ZappingOoki extends BaseSwapping {
+    protected multicallRouter!: MulticallRouter
+    protected userAddress!: string
+    protected iToken!: Ooki
+    protected iTokenAddress!: string
+
+    public async exactIn(
+        tokenAmountIn: TokenAmount,
+        iTokenAddress: string,
+        chainId: ChainId,
+        from: string,
+        to: string,
+        revertableAddress: string,
+        slippage: number,
+        deadline: number,
+        use1Inch = true
+    ): SwapExactIn {
+        this.multicallRouter = this.symbiosis.multicallRouter(chainId)
+        this.userAddress = to
+
+        this.iToken = this.symbiosis.ookiIToken(iTokenAddress, chainId)
+
+        const tokenAddress = await this.iToken.loanTokenAddress()
+        const tokenContract = new Contract(tokenAddress, ERC20, this.symbiosis.providers.get(chainId))
+        const decimals = await tokenContract.decimals()
+
+        const token = new Token({
+            address: tokenAddress,
+            chainId: chainId,
+            decimals,
+        })
+
+        return this.doExactIn(
+            tokenAmountIn,
+            token,
+            from,
+            this.multicallRouter.address,
+            revertableAddress,
+            slippage,
+            deadline,
+            use1Inch
+        )
+    }
+
+    protected finalReceiveSide(): string {
+        return this.multicallRouter.address
+    }
+
+    protected finalCalldata(): string | [] {
+        return this.buildMulticall()
+    }
+
+    protected finalOffset(): number {
+        return 36
+    }
+
+    protected swapTokens(): string[] {
+        const tokens = this.transit.route.map((i) => i.address)
+        if (this.tradeC) {
+            tokens.push(wrappedToken(this.tradeC.amountOut.token).address)
+        } else {
+            tokens.push(this.iTokenAddress)
+        }
+        return tokens
+    }
+
+    private buildMulticall() {
+        const callDatas = []
+        const receiveSides = []
+        const path = []
+        const offsets = []
+
+        let amount
+        let supplyToken
+
+        if (this.tradeC) {
+            amount = this.tradeC.tokenAmountIn.raw.toString()
+            supplyToken = this.tradeC.amountOut.token
+
+            callDatas.push(this.tradeC.callData)
+            receiveSides.push(this.tradeC.routerAddress)
+            path.push(this.tradeC.tokenAmountIn.token.address)
+            offsets.push(this.tradeC.callDataOffset!)
+        } else {
+            amount = this.transit.amountOut.raw.toString()
+            if (this.transit.direction === 'mint') {
+                supplyToken = this.transit.amountOut.token
+            } else {
+                supplyToken = this.transit.feeToken
+            }
+        }
+
+        const ookiCalldata = this.iToken.interface.encodeFunctionData('mint', [
+            this.userAddress,
+            supplyToken.toString(),
```

Comment:
You need to pass `amount` to mint, not a token address.

pass `this.transit.amountOut.raw.toString` here